src/test/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceServiceSpec.scala

```diff
@@ -1779,6 +1804,54 @@ class ResourceServiceSpec
     testDeleteResource(managedGroupResourceType)
   }
 
+  it should "delete any action managed identites for the resource while it deletes the resource" in {
+    assume(databaseEnabled, databaseEnabledClue)
+
+    val resource = FullyQualifiedResourceId(defaultResourceType.name, ResourceId("my-resource"))
+    // There's no actual need for it to be a real "billing profile", we just need a resource to attach the managed resource group to.
+    val billingProfileResource = FullyQualifiedResourceId(defaultResourceType.name, ResourceId(UUID.randomUUID().toString))
+    val ownerRoleActions = defaultResourceType.roles.find(_.roleName == defaultResourceType.ownerRoleName).get.actions
+
+    val managedResourceGroupCoordinates = ManagedResourceGroupCoordinates(
+      TenantId(UUID.randomUUID().toString),
+      SubscriptionId(UUID.randomUUID().toString),
+      ManagedResourceGroupName(UUID.randomUUID().toString)
+    )
+
+    val managedResourceGroup = ManagedResourceGroup(managedResourceGroupCoordinates, BillingProfileId(billingProfileResource.resourceId.value))
+
+    azuredService.createResourceType(defaultResourceType, samRequestContext).unsafeRunSync()
+    runAndWait(azuredService.createResource(defaultResourceType, resource.resourceId, dummyUser, samRequestContext))
+    runAndWait(azuredService.createResource(defaultResourceType, billingProfileResource.resourceId, dummyUser, samRequestContext))
+    runAndWait(azureManagedResourceGroupDAO.insertManagedResourceGroup(managedResourceGroup, samRequestContext))
+    ownerRoleActions.foreach { action =>
+      val ami = ActionManagedIdentity(
+        ActionManagedIdentityId(
+          resource,
+          action,
+          BillingProfileId(billingProfileResource.resourceId.value)
+        ),
+        ManagedIdentityObjectId(UUID.randomUUID().toString),
+        ManagedIdentityDisplayName(s"${resource.resourceId.value}-${action.value}"),
+        managedResourceGroupCoordinates
+      )
+      runAndWait(dirDAO.createActionManagedIdentity(ami, samRequestContext))
+    }
+    when(mockAzureService.deleteActionManagedIdentity(any[ActionManagedIdentityId], any[SamRequestContext])).thenReturn(IO.unit)
+
+    assert(dirDAO.getAllActionManagedIdentitiesForResource(resource, samRequestContext).unsafeRunSync().nonEmpty)
```

Comment:
I am confused shouldnt this return something at this line and then after deleting the resource not return anything?

Reply:
Ah, this passes because the test runs against a real database, but a fake azure service. I've re-arranged the lines to that it makes a little more sense.